### DISCUSSION TITLE
`dotnet new` templates

### DISCUSSION THREAD

**Discussion by tznind** [tui-cs/Terminal.Gui#1978] [Show and tell]
I've created a [template for gui.cs](https://github.com/tznind/gui.cs.template/blob/main/README.md)

I was investigating how to add new templates to `dotnet new`.  Turns out it is quite easy (following the [microsoft docs](https://docs.microsoft.com/en-us/dotnet/core/tools/custom-templates) and the [handy samples](https://github.com/dotnet/dotnet-template-samples) )

So now users can get started with gui.cs with 3 lines of code:

```
# Adds templates from NuGet
dotnet new --install Terminal.Gui.templates

# Use one of the templates
dotnet new tui -n myproj

# Build and run the project thats created
cd myproj
dotnet run
```

![image](https://user-images.githubusercontent.com/31306100/187677261-95e7b826-ec8a-421f-8fd3-18d214312a5d.png)

It seems pretty easy to add multiple templates to the same NuGet package so if we wanted later on we could add templates for multi threading etc or anthing else that is a little tricky to setup.

Also appears in the 'New=>Project' list after installing:
![image](https://user-images.githubusercontent.com/31306100/187680835-7f55c91a-ad7f-4b71-a163-b131b16ea6bb.png)

Let me know what you guys think.

**Comment by BDisp** [comment]
I like the templates idea.

**Comment by tig** [comment]
I'm in love with this.

**Comment by tznind** [comment]
Templates are now part of the official gui-cs org and have a new name of `Terminal.Gui.templates` and a new verb of `tui`!  See this issue: https://github.com/gui-cs/Terminal.Gui.templates/issues/4

```
dotnet new --install Terminal.Gui.templates

dotnet new tui -n myproj
cd myproj
dotnet run
```

**Reply by tig** [reply]
This is so trick.

However, for a new user, it's completely non-obvious how to quit the resulting app (ctrl-q).

Perhaps do this?

<img width="411" alt="image" src="https://user-images.githubusercontent.com/585482/196270682-6fcc142f-08ac-407b-9c0a-3f552c3f7a1c.png">

**Reply by tznind** [reply]
Good point! I've added it (https://github.com/gui-cs/Terminal.Gui.templates/commit/a25ac020fe11ac6430a07e7c3b71325926f223ab) and pushed as 1.0.3

I put it in the .Designer.cs file via TerminalGuiDesigner just to be consistent with the rest of the template.
